dev/Repeater/UniformGridLayout.cpp

```diff
@@ -213,10 +213,11 @@ winrt::Rect UniformGridLayout::Algorithm_GetExtent(
 
     if (itemsCount > 0)
     {
+        // Only use all of the space if item stretch is fill, otherwise size layout according to items placed
         extent.*MinorSize() =
-            std::isfinite(availableSizeMinor) ?
+            std::isfinite(availableSizeMinor) && m_itemsStretch == winrt::UniformGridLayoutItemsStretch::Fill ?
             availableSizeMinor :
-            std::max(0.0f, itemsCount * GetMinorSizeWithSpacing(context) - static_cast<float>(MinItemSpacing()));
+            std::max(0.0f, itemsPerLine * GetMinorSizeWithSpacing(context) - static_cast<float>(MinItemSpacing()));
```

Comment:
>0.0f [](start = 21, length = 4)

Should this be availableSizeMinor? @RanjeshJ

Reply:
This is correct. If there are any items present, we should size the extent based on the item sizes irrespective of what the available size that was provided by the parent. 

---
In reply to: [362570947](https://github.com/microsoft/microsoft-ui-xaml/pull/1802#discussion_r362570947) [](ancestors = 362570947)